include/git2/transport.h

```diff
@@ -287,6 +296,25 @@ GIT_EXTERN(int) git_cred_default_new(git_cred **out);
  */
 GIT_EXTERN(int) git_cred_username_new(git_cred **cred, const char *username);
 
+#ifdef GIT_SSH_MEMORY_CREDENTIALS
```

Comment:
We can't do this. The external API cannot be different according to the build flags. This function declaration must always be here. In case this is not available, the function body will return an error.